app/templates/application.hbs

```diff
@@ -2,7 +2,7 @@
 
 {{page-title "CodeCrafters"}}
 
-<div class="bg-gray-50 min-h-screen">
+<div class="application-container bg-gray-50 min-h-screen {{if this.darkMode.isEnabled 'dark'}}">
```

Comment:
Now that I look at this with fresh eyes, it feels a bit confusing that `darkMode.isEnabled` and the value from `{{is-dark-mode` can be different. The names don't convey that they check different things. Not sure of a better name though, one idea was to make the modifier `{{is-within-dark-area ...}}`.

Reply:
`did-load-dark-parents` would express exactly what it does, and it's a modifier, so the name has to be a verb...

Reply:
In general, I would ditch dark areas and just hardcode dark theme in them where needed, somehow.


Having to scan the DOM parents for `.dark` class makes things 9 levels more complicated: this is the only reason this modifier is needed. Otherwise the logic everywhere could have been as simple as:

```
<CodeMirror @theme={{this.darkMode.isEnabled 'githubDark' : 'githubLight'}}
```

No need for modifier + action to handle it's callback

Reply:
Discussed on slack - we're going forward with ditching the modifier and instead just hardcoding `@theme` for syntax highlighted diffs in dark areas